Postgresql sink constructs invalid upsert sql when pk.fields include all fields

io.confluent.connect.jdbc.sink.dialect.PostgreSqlDialect.getUpsertQuery()

change

    builder.append(") ON CONFLICT (");
    joinToBuilder(builder, ",", keyCols, escaper());
    builder.append(") DO UPDATE SET ");
    joinToBuilder(
        builder,
        ",",
        cols,
        new StringBuilderUtil.Transform<String>() {
          @Override
          public void apply(StringBuilder builder, String col) {
            builder.append(escaped(col)).append("=EXCLUDED.").append(escaped(col));
          }
        }
    );

to

    builder.append(") ON CONFLICT (");
    joinToBuilder(builder, ",", keyCols, escaper());
    if (cols.size() > 0) {
      builder.append(") DO UPDATE SET ");
      joinToBuilder(
          builder,
          ",",
          cols,
          new StringBuilderUtil.Transform<String>() {
            @Override
            public void apply(StringBuilder builder, String col) {
              builder.append(escaped(col)).append("=EXCLUDED.").append(escaped(col));
            }
          });
    } else {
      builder.append(") DO NOTHING");
    }

I am unable to get the build working even after following the steps in https://github.com/confluentinc/kafka-connect-jdbc/wiki/FAQ.

https://github.com/confluentinc/common fails to build with the following error

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] assembly-plugin-boilerplate ........................ SUCCESS [ 14.187 s]
[INFO] Build Tools ........................................ SUCCESS [ 12.950 s]
[INFO] common ............................................. SUCCESS [02:01 min]
[INFO] utils .............................................. SUCCESS [ 10.356 s]
[INFO] metrics ............................................ SUCCESS [ 10.492 s]
[INFO] config ............................................. SUCCESS [  5.404 s]
[INFO] common-logging ..................................... FAILURE [ 19.570 s]
[INFO] confluent-log4j-extensions ......................... SKIPPED
[INFO] confluent-log4j2-extensions ........................ SKIPPED
[INFO] package ............................................ SKIPPED
[INFO] Common ............................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:37 min
[INFO] Finished at: 2019-02-13T16:23:56+05:30
[INFO] Final Memory: 28M/300M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project common-logging: Could not resolve dependencies for
project io.confluent:common-logging:jar:5.3.0-SNAPSHOT: The following artifacts could not be
resolved: org.apache.kafka:connect-api:jar:2.3.0-SNAPSHOT, org.apache.kafka:connect-json:jar:2.3.0-SNAPSHOT:
Could not find artifact org.apache.kafka:connect-api:jar:2.3.0-SNAPSHOT in confluent (http://packages.confluent.io/maven/) -> [Help 1]
```

--- a/src/main/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java
+++ b/src/main/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java
@@ -243,11 +243,15 @@ public class PostgreSqlDatabaseDialect extends GenericDatabaseDialect {
            .delimitedBy(",")
            .transformedBy(ExpressionBuilder.columnNames())
            .of(keyColumns);
-    builder.append(") DO UPDATE SET ");
-    builder.appendList()
-           .delimitedBy(",")
-           .transformedBy(transform)
-           .of(nonKeyColumns);
+    if (nonKeyColumns.isEmpty()) {
+      builder.append(") DO NOTHING");
+    } else {
+      builder.append(") DO UPDATE SET ");
+      builder.appendList()
+              .delimitedBy(",")
+              .transformedBy(transform)
+              .of(nonKeyColumns);
+    }
     return builder.toString();
   }
 

--- a/src/test/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialectTest.java
+++ b/src/test/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialectTest.java
@@ -239,6 +239,16 @@ public class PostgreSqlDatabaseDialectTest extends BaseDialectTest<PostgreSqlDat
         )
     );
 
+    assertEquals(
+            "INSERT INTO \"Customer\" (\"id\",\"name\",\"salary\",\"address\") " +
+                    "VALUES (?,?,?,?) ON CONFLICT (\"id\",\"name\",\"salary\",\"address\") DO NOTHING",
+            dialect.buildUpsertQueryStatement(
+                    customer,
+                    columns(customer, "id", "name", "salary", "address"),
+                    columns(customer)
+            )
+    );
+
     quoteIdentfiiers = QuoteMethod.NEVER;
     dialect = createDialect();
 
@@ -251,6 +261,16 @@ public class PostgreSqlDatabaseDialectTest extends BaseDialectTest<PostgreSqlDat
             columns(customer, "id"),
             columns(customer, "name", "salary", "address")
         )
+    );
+
+    assertEquals(
+            "INSERT INTO Customer (id,name,salary,address) " +
+                    "VALUES (?,?,?,?) ON CONFLICT (id,name,salary,address) DO NOTHING",
+            dialect.buildUpsertQueryStatement(
+                    customer,
+                    columns(customer, "id", "name", "salary", "address"),
+                    columns(customer)
+            )
     );
   }
 

